### PR TITLE
added docker setings in build for kafkaEngineRuntime + added entrypoint script

### PR DESCRIPTION
**Pull request by maciej-brzezinski**
a draft for building a docker image for kafkaEngineRuntime

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -851,10 +853,29 @@ lazy val nuKafkaEngineBinTest: Project = (project in engine("base/nu-kafka-engin
       kafkaBaseEngineRuntime / Universal / stage
     ).value,
     libraryDependencies ++= Seq(
-      "commons-io" % "commons-io" % commonsIOV
+      "commons-io" % "commons-io" % commonsIOV,
+      "com.whisk" %% "docker-testkit-scalatest" % "0.9.0" % "it",
+      "com.whisk" %% "docker-testkit-impl-spotify" % "0.9.0" % "it",
```
Comment (mproch) [inline build.sbt:858]: we don't need those two anymore

File: engine/base/kafka/src/universal/bin/nu-kafka-engine-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+NUSSKNACKER_DIR=`dirname "$0" | xargs -I{} readlink -f {}/..`
+CONF_DIR="$NUSSKNACKER_DIR/conf"
+LIB_DIR="$NUSSKNACKER_DIR/lib"
+export COMPONENTS_DIR="$NUSSKNACKER_DIR/components"
+
+CLASSPATH=${CLASSPATH:-$LIB_DIR/*}
+CONFIG_FILE=${CONFIG_FILE-"$CONF_DIR/application.conf"}
+SCENARIO_FILE=${SCENARIO_FILE-"$CONF_DIR/scenario.json"}
+LOGBACK_FILE=${LOGBACK_FILE-"$CONF_DIR/docker-logback.xml"}
```
Comment (mproch) [inline engine/base/kafka/src/universal/bin/nu-kafka-engine-entrypoint.sh:13]: we don't have docker-logback.xml - let's use logback.xml

File: build.sbt [hunk 3]
```diff
@@ -865,9 +868,25 @@ lazy val liteKafkaEngineApi = (project in engine("base/kafka-api")).
     )
   ).dependsOn(api)
 
+lazy val liteKafkaEngineDockerSettings = {
+  val workingDir = "/opt/nussknacker"
+
+  commonDockerSettings ++ Seq(
+    dockerEntrypoint := Seq(s"$workingDir/bin/nu-kafka-engine-entrypoint.sh"),
+    dockerExposedPorts := Seq(dockerPort), //to consider if it needs a seperate port
+    dockerExposedVolumes := Seq(s"$workingDir/storage"),
```
Comment (dswiecki) [inline build.sbt:877]: is it necessary ?

File: build.sbt [hunk 4]
```diff
@@ -865,9 +868,24 @@ lazy val liteKafkaEngineApi = (project in engine("base/kafka-api")).
     )
   ).dependsOn(api)
 
+lazy val liteKafkaEngineDockerSettings = {
+  val workingDir = "/opt/nussknacker"
+
+  commonDockerSettings ++ Seq(
+    dockerEntrypoint := Seq(s"$workingDir/bin/nu-kafka-engine-entrypoint.sh"),
+    dockerExposedPorts := Seq(dockerPort),
```
Comment (mproch) [inline build.sbt:876]: do we need it?

File: build.sbt [hunk 5]
```diff
@@ -1258,7 +1276,7 @@ lazy val root = (project in file("."))
       tagRelease,
       releaseStepCommandAndRemaining("+publishSigned"),
       releaseStepCommand("dist/Universal/packageZipTarball"),
-      releaseStepCommand("kafkaBaseEngineRuntime/Universal/packageZipTarball"),
+      releaseStepCommand("liteKafkaEngineRuntime/Universal/packageZipTarball"),
```
Comment (mproch) [inline build.sbt:1279]: I think we also need Docker/publish here?
Also, we probably want to publish image in preview branch build? pr.yml, link 579?

File: engine/base/kafka-runtime-bin-test/src/it/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineBinTest.scala [hunk 6]
```diff
@@ -25,7 +25,7 @@ class NuKafkaEngineBinTest extends FunSuite with KafkaSpec with Matchers with La
   private val outputTopic = "output"
 
   // FIXME: fix NoClassDefFoundError: org/apache/flink/streaming/api/functions/sink/SinkFunction
```
Comment (mproch) [inline engine/base/kafka-runtime-bin-test/src/it/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineBinTest.scala:27]: I don't think we're done with this test yet ;)

File: engine/base/kafka-runtime-bin-test/src/it/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineDockerTest.scala [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.dimafeng.testcontainers.{FixedHostPortGenericContainer, ForAllTestContainer}
+import org.scalatest.{BeforeAndAfter, FunSuite, Matchers}
+import org.testcontainers.containers.BindMode
+import pl.touk.nussknacker.engine.kafka.KafkaSpec
+import pl.touk.nussknacker.test.VeryPatientScalaFutures
+
+class NuKafkaEngineDockerTest extends FunSuite with ForAllTestContainer with KafkaSpec  with VeryPatientScalaFutures with Matchers with BeforeAndAfter {
+
+  private val nuEngineRuntimeDockerName = "touk/nussknacker-lite-kafka-runtime:1.0.1-SNAPSHOT"
```
Comment (mproch) [inline engine/base/kafka-runtime-bin-test/src/it/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineDockerTest.scala:11]: It won't work (at least not in CI...) We have to publish image locally I think?

File: engine/base/kafka/src/universal/conf/application.conf [hunk 8]
```diff
@@ -1,4 +1,10 @@
 modelConfig: {
-  classPath: ["model/liteModel.jar", "components/baseComponents.jar"]
+  classPath: ["model/liteModel.jar", "components/liteBaseComponents.jar"]
   classPath: ${?MODEL_CLASS_PATH}
+  kafka {
+    kafkaAddress: "kafka:9092"
```
Comment (dswiecki) [inline engine/base/kafka/src/universal/conf/application.conf:5]: we should use environment variable here for kafkaAddress

File: engine/base/kafka/src/universal/conf/application.conf [hunk 9]
```diff
@@ -1,4 +1,10 @@
...
+  }
+  exceptionHandlingConfig {
+  topic: "test"
```
Comment (dswiecki) [inline engine/base/kafka/src/universal/conf/application.conf:8]: and here

**Pull request merged**